### ISSUE TITLE
Add optional host filtering step at beginning

### ISSUE THREAD

**Issue by niemasd** [niemasd/MultiVirusConsensus#1]
Before running Minimap2 on the viral genomes, I should add an optional first host filtering step. I want to use a Bloom Filter based host filtering tool (or similar data structure) for the sake of speed and low memory since we don't care about the alignments to the host genome

BioBloom seems promising. Doing literature search to see if there are any newer / better tools

https://github.com/BCGSC/biobloom

http://doi.org/10.1093/bioinformatics/btu558

**Comment by niemasd** [comment]
* Vector Quotient Filter - https://doi.org/10.1145/3448016.3452841
  * No code, so maybe not this paper (but maybe another paper)
* HoCoRT - https://doi.org/10.1186/s12859-023-05492-w
  * https://github.com/ignasrum/hocort
  * It seems like this tool just wraps around other tools, including BioBloom, so maybe BioBloom is the best pick
  * In the accuracy + runtime comparison, the BioBloom mode seems best in terms of both
* FastqPuri - https://doi.org/10.1186/s12859-019-2799-0
  * Seems like it just wraps around BioBloom and similar tools, so maybe BioBloom is the best pick
* Adaptive Quotient Filter - https://doi.org/10.1145/3677128
  * No code, so maybe not this paper (but maybe another paper)
* Multi-index Bloom Filter - https://doi.org/10.1073/pnas.1903436117
  * Seems like this is part of BioBloom, so maybe BioBloom is the best pick
* TAP - https://doi.org/10.1186/s12920-018-0402-6
  * Seems like this is part of BioBloom, so maybe BioBloom is the best pick

**TL;DR: Looks like BioBloom is the best pick**

**Comment by niemasd** [comment]
Added in https://github.com/niemasd/MultiVirusConsensus/commit/29089c80d33fce8f61c309b6986ec6ddb575d504
